feat(helm): add Arda Credit Helm chart for deploying the application …

aWN4Y25pa2EK opened the pull request:
…on Kubernetes

- Create a new Helm chart for the Arda Credit application, which includes deployment configurations, service definitions, and necessary resources.
- Include PostgreSQL as a dependency for easy integration and management.
- Provide configuration options for autoscaling, high availability, and security best practices.
- Add comprehensive documentation in the README for installation and usage instructions.
- Implement necessary Kubernetes resources such as Deployments, Services, ConfigMaps, Secrets, and Ingress for a complete deployment setup.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- New Features
  - Introduced a Helm chart for deploying Arda Credit with configurable app and server components, autoscaling (HPA), high availability (PDBs, anti-affinity), RBAC/service account, Secrets, ConfigMap-driven settings, Services, and optional Ingress.
  - Bundled PostgreSQL dependency with option to use an external database.
  - Namespace creation supported via pre-install/upgrade hook.

- Documentation
  - Added comprehensive Helm chart README with setup guides, EKS ingress examples, scaling, security, upgrades, and troubleshooting.

- Chores
  - Updated ignore rules and added Helm-specific .helmignore to exclude build and development artifacts.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

aWN4Y25pa2EK committed:
feat(helm): add Arda Credit Helm chart for deploying the application on Kubernetes

- Create a new Helm chart for the Arda Credit application, which includes
 deployment configurations, service definitions, and necessary resources.
- Include PostgreSQL as a dependency for easy integration and management.
- Provide configuration options for autoscaling, high availability, and security
 best practices.
- Add comprehensive documentation in the README for installation and usage
 instructions.
- Implement necessary Kubernetes resources such as Deployments, Services,
 ConfigMaps, Secrets, and Ingress for a complete deployment setup.

MSevey commented on helm/charts/arda-credit/values.yaml:
```diff
@@ -0,0 +1,248 @@
+# Application secrets - these should be overridden in environment-specific values
+secrets:
+  app:
+    sindriApiKey: "your-sindri-api-key"
```
secrets.SINDRI_API_KEY is already set for the repo. 

I also sent you an invite for the sindri team so you can create your own key for testing.

aWN4Y25pa2EK committed:
feat(helm): update security context references in deployment templates for app and server
feat(helm): add support for app-specific security contexts in values.yaml
fix(helm): correct image registry references for app and server images
feat(helm): configure DATABASE_URL environment variable for PostgreSQL
chore(helm): reduce replica count for app and server deployments to 1
chore(helm): add Kind cluster configuration for testing Arda Credit Helm chart

aWN4Y25pa2EK committed:
feat(helm): add global nodeSelector, tolerations, and affinity settings for better resource management
refactor(helm): consolidate nodeSelector, tolerations, and affinity logic in app and server deployment templates to reduce redundancy

aWN4Y25pa2EK committed:
chore(.gitignore): add Helm chart dependencies to .gitignore to prevent tracking of downloaded chart archives
chore(.helmignore): add empty line for better readability
chore(charts): remove outdated PostgreSQL chart archive to clean up unused files

aWN4Y25pa2EK committed:
feat(helm-release): add GitHub Actions workflow for Helm chart release to automate deployment process
fix(Chart.yaml): update chart version from 0.1.0 to 0.2.0 to reflect new changes
fix(server-deployment.yaml): update DATABASE_URL to use environment variable for password for better security
docs(README.md): add EKS deployment instructions for AWS Load Balancer Controller and ACM certificate setup
chore(values.yaml): update ingress host to app.stag.aws.arda.xyz for staging environment and add comments for ALB configuration options

aWN4Y25pa2EK committed:
feat(helm-release): add infra/k8s branch to trigger Helm release workflow on push

aWN4Y25pa2EK committed:
chore(chart.yaml): bump version from 0.2.0 to 0.3.0 to reflect new release

aWN4Y25pa2EK committed:
chore(helm-release): remove --validate flag from helm template command for simplicity
chore(Chart.yaml): bump version from 0.3.0 to 0.4.0 to reflect new changes

aWN4Y25pa2EK committed:
chore: remove Helm release workflow and related files to streamline CI/CD process
fix(values.yaml): change image pull policy to Always for consistent deployments
fix(values.yaml): change rustLog level from info to debug for better logging during development

MSevey committed:
Merge branch 'main' into infra/k8s
